Prefer default allocator over nullptr

LumioseSil opened the pull request:
#### 0bd425f2bd48d64caffb4f3fcc63b2b126c7a98c
<pre>
Prefer default allocator over nullptr
<a href="https://bugs.webkit.org/show_bug.cgi?id=252169">https://bugs.webkit.org/show_bug.cgi?id=252169</a>

Reviewed by NOBODY (OOPS!).

Prefer the default allocate over nullptr when passing to CF
allocation functions.

*Source\WebCore\platform\LocalizedStrings.cpp:
*Source\WebKitLegacy\mac\Misc\WebLocalizableStrings.mm:
*Source\WebKitLegacy\mac\Misc\WebNSDataExtras.mm:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0ca14f0ceb4b46c55a4ca726aee51ba6f72d2c2e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107768 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16824 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40665 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116913 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/116294 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111658 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/18238 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8140 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99947 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113525 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13770 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96969 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/41471 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95677 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28610 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/83313 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9788 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29958 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10475 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6852 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15942 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49542 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12038 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/platform/LocalizedStrings.cpp:
```diff
@@ -67,7 +67,7 @@ ALLOW_NONLITERAL_FORMAT_END
 {
 #if USE(CF) && PLATFORM(WIN)
-    auto cfFormat = adoptCF(CFStringCreateWithCStringNoCopy(nullptr, format, kCFStringEncodingUTF8, kCFAllocatorNull));
+    auto cfFormat = adoptCF(CFStringCreateWithCStringNoCopy(kCFAllocatorDefault, format, kCFStringEncodingUTF8, kCFAllocatorNull));
```
This does not really seem like a style improvement, more like churn to me.

darinadler commented on Source/WebKitLegacy/mac/Misc/WebNSDataExtras.mm:
```diff
@@ -83,9 +83,9 @@ - (NSString *)_web_capitalizeRFC822HeaderFieldName
     if (somethingChanged) {
         if (useUniCharPtr)
-            result = adoptCF(CFStringCreateWithCharactersNoCopy(kCFAllocatorDefault, uniCharPtr, len, nullptr)).bridgingAutorelease();
+            result = adoptCF(CFStringCreateWithCharactersNoCopy(kCFAllocatorDefault, uniCharPtr, len, kCFAllocatorDefault)).bridgingAutorelease();
```
The description here is inaccurate. This change does not fix a bug. The backing buffer is correctly deallocated by the old code. The constant `kCFAllocatorDefault` simply expands to `nullptr`. So the code here is identical, and this is just a style change. There would be a bug here if this was using `kCFAllocatorNull`, which is not equal to `nullptr`, but it’s not.

So I understand if you want to suggest a style change. But it’s not appropriate to land that in a patch that claims to "deallocate the backing buffer".

AZero13 replied:
I had no idea. I thought kCFAllocatorNull WAS nullptr.

darinadler replied:
Not to preach, but this is a reason testing is important, even if we think we spotted a mistake by code inspection.

LumioseSil committed:
Prefer kCFAllocatorDefault over nullptr
https://bugs.webkit.org/show_bug.cgi?id=252169

Reviewed by NOBODY (OOPS!).

Prefer the default allocate over nullptr when passing to CF
allocation functions.

*Source\WebCore\platform\LocalizedStrings.cpp:
*Source\WebKitLegacy\mac\Misc\WebLocalizableStrings.mm:
*Source\WebKitLegacy\mac\Misc\WebNSDataExtras.mm:
